prov/efa/src/rdm/efa_rdm_cq.c

```diff
@@ -120,12 +144,20 @@ void efa_rdm_cq_proc_ibv_recv_rdma_with_imm_completion(
 
 	efa_cntr_report_rx_completion(&ep->base_ep.util_ep, flags);
 
-	/* Recv with immediate will consume a pkt_entry, but the pkt is not
-	   filled, so free the pkt_entry and record we have one less posted
-	   packet now. */
-	assert(pkt_entry);
-	ep->efa_rx_pkts_posted--;
-	efa_rdm_pke_release_rx(pkt_entry);
+	/**
+	 * For unsolicited wc, pkt_entry can be NULL, so we can only
+	 * access it for solicited wc.
+	 */
+	if (!efa_rdm_cq_wc_is_unsolicited(ibv_cq_ex)) {
+		/**
+		 * Recv with immediate will consume a pkt_entry, but the pkt is not
+		 * filled, so free the pkt_entry and record we have one less posted
+		 * packet now.
+		 */
+		assert(pkt_entry);
+		ep->efa_rx_pkts_posted--;
+		efa_rdm_pke_release_rx(pkt_entry);
+	}
```

Comment:
Is this block supposed to be guarded with a directive? E.g. `#if HAVE_CAPS_UNSOLICITED_WRITE_RECV`

Reply:
efa_rdm_cq_wc_is_unsolicited is defined for both `HAVE_CAPS_UNSOLICITED_WRITE_RECV=1` and `HAVE_CAPS_UNSOLICITED_WRITE_RECV=0`. For the latter one, it will just return false